sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/json/JsonFileFormat.scala

```diff
@@ -114,6 +114,21 @@ class JsonFileFormat extends TextBasedFileFormat with DataSourceRegister {
     }
 
     (file: PartitionedFile) => {
+      // SPARK-21610: when the `requiredSchema` only contains `_corrupt_record`,
+      // the derived `actualSchema` is empty and the `_corrupt_record` are all null for all rows.
+      // When users requires only `_corrupt_record`, we assume that the corrupt records are required
+      // for all json fields, i.g., all items in dataSchema.
```

Comment:
This comment is wrong now. We can get rid of this comment as the following exception is self-explained.